File: admin/server/router/ddz/user_account.go
package ddz

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type DDZUserAccountRouter struct{}

func (r *DDZUserAccountRouter) InitDDZUserAccountRouter(Router *gin.RouterGroup) {
	ddzUserAccountRouter := Router.Group("ddz/userAccount").Use(middleware.OperationRecord())
	ddzUserAccountRouterWithoutRecord := Router.Group("ddz/userAccount")

	ddzUserAccountApi := v1.ApiGroupApp.DDZApiGroup.DDZUserAccountApi
	{
		ddzUserAccountRouter.POST("create", ddzUserAccountApi.CreateUserAccount)           // 创建用户账户
		ddzUserAccountRouter.DELETE("delete", ddzUserAccountApi.DeleteUserAccount)         // 删除用户账户
		ddzUserAccountRouter.PUT("update", ddzUserAccountApi.UpdateUserAccount)            // 更新用户账户
		ddzUserAccountRouter.POST("status", ddzUserAccountApi.UpdateUserAccountStatus)     // 更新账户状态
		ddzUserAccountRouter.POST("bindPhone", ddzUserAccountApi.BindPhone)                // 绑定手机号
		ddzUserAccountRouter.POST("unbindWechat", ddzUserAccountApi.UnbindWeChat)          // 解绑微信
		ddzUserAccountRouter.POST("resetToken", ddzUserAccountApi.ResetToken)              // 重置Token(强制下线)
	}
	{
		ddzUserAccountRouterWithoutRecord.POST("list", ddzUserAccountApi.GetUserAccountList)     // 分页获取用户账户列表
		ddzUserAccountRouterWithoutRecord.GET("info", ddzUserAccountApi.GetUserAccountByID)      // 获取用户账户信息
		ddzUserAccountRouterWithoutRecord.POST("loginLog", ddzUserAccountApi.GetLoginLogList)    // 获取登录日志列表
	}
}

File: admin/web/src/api/ddz/userAccount.js
import adminService from '@/utils/adminRequest'

// 获取用户账户列表
export const getUserAccountList = (data) => {
  return adminService({
    url: '/ddz/userAccount/list',
    method: 'post',
    data
  })
}

// 获取用户账户信息
export const getUserAccountInfo = (id) => {
  return adminService({
    url: '/ddz/userAccount/info',
    method: 'get',
    params: { id }
  })
}

// 创建用户账户
export const createUserAccount = (data) => {
  return adminService({
    url: '/ddz/userAccount/create',
    method: 'post',
    data
  })
}

// 删除用户账户
export const deleteUserAccount = (data) => {
  return adminService({
    url: '/ddz/userAccount/delete',
    method: 'delete',
    data
  })
}

// 更新用户账户
export const updateUserAccount = (data) => {
  return adminService({
    url: '/ddz/userAccount/update',
    method: 'put',
    data
  })
}

// 更新用户账户状态
export const updateUserAccountStatus = (data) => {
  return adminService({
    url: '/ddz/userAccount/status',
    method: 'post',
    data
  })
}

// 绑定手机号
export const bindPhone = (data) => {
  return adminService({
    url: '/ddz/userAccount/bindPhone',
    method: 'post',
    data
  })
}

// 解绑微信
export const unbindWechat = (data) => {
  return adminService({
    url: '/ddz/userAccount/unbindWechat',
    method: 'post',
    data
  })
}

// 重置Token(强制下线)
export const resetToken = (data) => {
  return adminService({
    url: '/ddz/userAccount/resetToken',
    method: 'post',
    data
  })
}

// 获取登录日志列表
export const getLoginLogList = (data) => {
  return adminService({
    url: '/ddz/userAccount/loginLog',
    method: 'post',
    data
  })
}

// 检查手机号是否已存在
export const checkPhoneExists = (phone, excludeId) => {
  return adminService({
    url: '/ddz/userAccount/checkPhone',
    method: 'get',
    params: { phone, excludeId }
  })
}
